### PR TITLE
updating breakdown-tree for core@3.7 and appui@4.0

### PR DESCRIPTION
**Pull request by Jake-Screen**

### CODE REVIEW

**Review by grigasp — COMMENTED** [review]
Any reason not to add a dependency on `tree-widget-react` and use the visibility tree APIs from there?

File: apps/test-viewer/package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "@itwin/appui-abstract": "^3.7.1",
     "@itwin/appui-layout-react": "^4.0.0-dev.15",
     "@itwin/appui-react": "^4.0.0-dev.15",
+    "@itwin/breakdown-trees-react": "workspace:*",
```
Comment (aruniverse) [inline apps/test-viewer/package.json:10]: take a look at the readme for the viewer

you need to add the depedendency meta

just like measure-tools

File: packages/itwin/breakdown-trees/src/Views/IconButton.scss [hunk 2]
```diff
@@ -16,15 +15,15 @@
   background-color: transparent;
   outline: none;
   box-shadow: none;
-  font-family: $uicore-font-family;
-  font-size: $uicore-font-size;
+  font-family: var(--iui-font-sans);
+  font-size: calc(var(--iui-size-s) + var(--iui-size-3xs));
```
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/Views/IconButton.scss:19]: this line looks super sus, this cant be right

File: packages/itwin/breakdown-trees/src/Views/SearchBar/SearchBar.scss [hunk 3]
```diff
@@ -48,15 +46,15 @@
...
+    padding: var(--iui-size-s);
   }
 
   // override of the search box styling
   .search-bar-search-icons {
-    right: 0;
     opacity: 0;
-    color: $buic-icon-color;
     transition: all 0.2s ease;
+    color: var(--iui-color-text);
```
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/Views/SearchBar/SearchBar.scss:56]: doesnt look right

File: packages/itwin/breakdown-trees/src/BreakdownTrees.ts [hunk 4]
```diff
@@ -8,23 +8,27 @@
...
   private static _i18n?: Localization;
 
   /**
    * Called by IModelApp to initialize the BreakdownTrees
    * @param localization The internationalization service created by the IModelApp.
    */
-  public static async initialize(localization: Localization): Promise<void> {
-    BreakdownTrees._i18n = localization;
+  public static async initialize(localization?: Localization): Promise<void> {
+    if (this._initialized) return;
```
Comment (grigasp) [inline packages/itwin/breakdown-trees/src/BreakdownTrees.ts:28]: Please see point no. 7 under [Make judicious use of vertical screen space section](https://github.com/iTwin/itwinjs-core/blob/master/docs/learning/guidelines/typescript-coding-guidelines.md#make-judicious-use-of-vertical-screen-space)

File: packages/itwin/breakdown-trees/src/test/TreeComponents.test.tsx [hunk 5]
```diff
@@ -116,6 +117,17 @@ describe("TreeComponent tests.", () => {
...
+    const propertyValue = moq.Mock.ofType<PropertyValue>();
+    const propertyDescription = moq.Mock.ofType<PropertyDescription>();
+    propertyValue.setup((x) => x.valueFormat).returns(() => 0);
+    propertyDescription.setup((x) => x.name).returns(() => "propertyRecName");
+    propertyDescription.setup((x) => x.displayLabel).returns(() => "propertyRecDisplayLabel");
+    propertyDescription.setup((x) => x.typename).returns(() => "propertyRecTypeName");
+    propertyRecord.setup((x) => x.value).returns(() => propertyValue.object);
+    propertyRecord.setup((x) => x.property).returns(() => propertyDescription.object);
+    node.setup((x) => x.depth).returns(() => 1);
+    node.setup((x) => x.label).returns(() => propertyRecord.object);
```
Comment (grigasp) [inline packages/itwin/breakdown-trees/src/test/TreeComponents.test.tsx:130]: I think simply creating a `PropertyRecord` through its constructor or `PropertyRecord.fromString` would be easier more readable.

File: packages/itwin/breakdown-trees/src/test/Utils.tsx [hunk 6]
```diff
@@ -22,12 +22,15 @@ import { PropertyRecord } from "@itwin/appui-abstract";
...
 import { TreeWithRulesetControllerBase } from "../Controllers/TreeWithRulesetControllerBase";
 import type { Localization } from "@itwin/core-common";
+import { configureStore } from "@reduxjs/toolkit";
 
 function createAppStore(): Store {
   const rootReducer = combineReducers({
     FrameworkReducer,
   } as any);
 
+  // return configureStore({ reducer: rootReducer })
```
Comment (grigasp) [inline packages/itwin/breakdown-trees/src/test/Utils.tsx:32]: needs cleanup

File: packages/itwin/breakdown-trees/src/test/Utils.tsx [hunk 7]
```diff
@@ -53,9 +56,8 @@ export class TestUtils {
       // This is required by our I18n module (specifically the i18next package).
       (global as any).XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest; // eslint-disable-line @typescript-eslint/no-var-requires
       this.store = createAppStore();
-
       await UiFramework.initialize(this.store);
-      console.log("HELLO");
+      // await UiFramework.initialize(this.store);
```
Comment (grigasp) [inline packages/itwin/breakdown-trees/src/test/Utils.tsx:60]: needs cleanup

File: packages/itwin/breakdown-trees/src/test/functionalityProviders/BuildingClipPlanesProvider.test.ts [hunk 8]
```diff
@@ -45,6 +46,7 @@ describe("BuildingClipPlanesProvider", () => {
     queryBuildingRangeStub.restore();
     createRange3DPlaneStub.restore();
     TestUtils.terminateUiFramework();
+    BreakdownTrees.terminate();
```
Comment (grigasp) [inline packages/itwin/breakdown-trees/src/test/functionalityProviders/BuildingClipPlanesProvider.test.ts:49]: shouldn't the `initialize` call be somewhere around too?

File: packages/itwin/breakdown-trees/src/test/functionalityProviders/SpaceClipPlanesProvider.test.ts [hunk 9]
```diff
@@ -30,7 +30,11 @@ describe("SpaceClipPlanesProvider", () => {
   const dataProviderMock = moq.Mock.ofType<IPresentationTreeDataProvider>();
 
   before(async () => {
-    await TestUtils.initializeUiFramework(connection.object);
+    try {
+      await TestUtils.initializeUiFramework(connection.object);
+    } catch (err) {
+      console.log(err);
+    }
```
Comment (grigasp) [inline packages/itwin/breakdown-trees/src/test/functionalityProviders/SpaceClipPlanesProvider.test.ts:37]: Why do we need try/catch here? This suggests that "terminate" call is missing somewhere.

File: apps/test-viewer/src/components/Viewer.tsx [hunk 10]
```diff
@@ -22,6 +24,7 @@ export function Viewer() {
   const { iTwinId, iModelId } = useIModelInfo();
 
   return (
+
```
Comment (grigasp) [inline apps/test-viewer/src/components/Viewer.tsx:27]: unnecessary whitespace

File: apps/test-viewer/src/components/Viewer.tsx [hunk 11]
```diff
@@ -9,10 +9,12 @@ import { Viewer as WebViewer, ViewerPerformance } from "@itwin/web-viewer-react"
 import { history } from "../history";
 import { getUiProvidersConfig } from "../UiProvidersConfig";
 import { useAuthorizationContext } from "./Authorization";
+import { BreakdownTrees } from "@itwin/breakdown-trees-react";
 
 const uiConfig = getUiProvidersConfig();
 
 async function onIModelAppInit() {
+  await BreakdownTrees.initialize();
```
Comment (grigasp) [inline apps/test-viewer/src/components/Viewer.tsx:17]: I think the plan was to initialize all components within their widget's initialization function rather than at the app level. E.g. in `UiProvidersConfig.tsx` change the tree widget's config to:
```tsx
  [
    "tree-widget",
    {
      initialize: async () => {
        await TreeWidget.initialize();
        await BreakdownTrees.initialize();
      },
      createUiItemsProvider: () => new TreeWidgetUiItemsProvider({
        additionalTrees: [getSpatialTree()]
      }),
    }
  ],
```

Adding @saskliutas for his opinion.

**Review by grigasp — CHANGES_REQUESTED** [review]

File: apps/test-viewer/src/UiProvidersConfig.tsx [hunk 12]
```diff
@@ -44,12 +49,42 @@ interface UiItem {
...
+      setGroupByDiscipline={() => { return "group-by-discipline" }}
+      displayGuids={true}
+      setIsDisplayGuids={() => { return "discard-guid-from-labe" }}
+      enableVisibility={true}
+      clipHeight={1.2}
+      clipAtSpaces={true}
+    />}
+  </>
+}
+const getSpatialTree = (): SelectableContentDefinition => {
```
Comment (saskliutas) [inline apps/test-viewer/src/UiProvidersConfig.tsx:70]: I think this function can be removed and object returned by this function can be inlined in `additionalTrees` array.

File: apps/test-viewer/src/UiProvidersConfig.tsx [hunk 13]
```diff
@@ -48,8 +49,19 @@ const configuredUiItems = new Map<string, UiItem>([
   [
     "tree-widget",
     {
-      initialize: async () => TreeWidget.initialize(),
-      createUiItemsProvider: () => new TreeWidgetUiItemsProvider(),
+      initialize: async () => {
+        TreeWidget.initialize();
```
Comment (aruniverse) [inline apps/test-viewer/src/UiProvidersConfig.tsx:53]: dont these need to be awaited?

File: apps/test-viewer/src/components/Viewer.tsx [hunk 14]
```diff
@@ -9,6 +9,7 @@ import { Viewer as WebViewer, ViewerPerformance } from "@itwin/web-viewer-react"
 import { history } from "../history";
 import { getUiProvidersConfig } from "../UiProvidersConfig";
 import { useAuthorizationContext } from "./Authorization";
+import { BreakdownTrees } from "@itwin/breakdown-trees-react";
```
Comment (aruniverse) [inline apps/test-viewer/src/components/Viewer.tsx:12]: revert

File: packages/itwin/breakdown-trees/src/BreakdownTrees.ts [hunk 15]
```diff
@@ -8,23 +8,28 @@
...
    * Called by IModelApp to initialize the BreakdownTrees
    * @param localization The internationalization service created by the IModelApp.
    */
-  public static async initialize(localization: Localization): Promise<void> {
-    BreakdownTrees._i18n = localization;
+  public static async initialize(localization?: Localization): Promise<void> {
+    if (this._initialized)
+      return;
+    // const local = new ITwinLocalization({ urlTemplate: `${window.location.origin}/locales/{{lng}}/{{ns}}.json` })
+    this._initialized = true;
```
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/BreakdownTrees.ts:31]: should set this after registering the namespace

File: packages/itwin/breakdown-trees/src/Views/NodeRenderers/FunctionalTreeNodeRenderer.module.scss [hunk 16]
```diff
@@ -2,28 +2,30 @@
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
+@import "~@itwin/itwinui-variables";
```
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/Views/NodeRenderers/FunctionalTreeNodeRenderer.module.scss:5]: i dont know if you need this import

**Review by aruniverse — COMMENTED** [review]
please review

File: apps/test-viewer/src/UiProvidersConfig.tsx (hunk 13 above)
Comment (Jake-Screen) [reply apps/test-viewer/src/UiProvidersConfig.tsx:53]: they are once they are once passed into the getUiProvidersConfig function

Comment (aruniverse) [reply apps/test-viewer/src/UiProvidersConfig.tsx:53]: refer to https://github.com/iTwin/viewer-components-react/pull/427#discussion_r1170802779

File: apps/test-viewer/src/UiProvidersConfig.tsx [hunk 17]
```diff
@@ -48,21 +49,35 @@ const configuredUiItems = new Map<string, UiItem>([
...
       createUiItemsProvider: () => new PropertyGridUiItemsProvider(),
     }
   ],
   [
     "measure-tools",
     {
-      initialize: async () => MeasureTools.startup(),
+      initialize: async () => {
+        await MeasureTools.startup();
+        MeasurementActionToolbar.setDefaultActionProvider();
```
Comment (aruniverse) [inline apps/test-viewer/src/UiProvidersConfig.tsx:79]: what is this?

File: packages/itwin/breakdown-trees/src/Views/ClassificationsTree.tsx [hunk 18]
```diff
@@ -46,7 +46,7 @@ export const ClassificationsTree: React.FC<ClassificationsTreeProps> = (props: C
     const functionIconMapperInner = props.additionalFunctionIconMapper ?? new TreeNodeFunctionIconInfoMapper(dataProvider);
     const optionItemsInner: OptionItemHandler[] = [];
     populateMapWithCommonMenuItems(CLASSIFICATIONS_TREE_NAME, functionIconMapperInner, dataProvider, classificationRules.id, props.eventHandlers);
-    const displayGuidHandlerInner = new GenericOptionItemHandler("Show Guids", BreakdownTrees.translate("contextMenu.showGuids"), "icon-label", () => { return props.displayGuids; }, props.setIsDisplayGuids);
+    const displayGuidHandlerInner = new GenericOptionItemHandler("Show Guids", BreakdownTrees.translate("contextMenu.showGuids"), "icon - label", () => { return props.displayGuids; }, props.setIsDisplayGuids);
```
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/Views/ClassificationsTree.tsx:49]: why?

File: packages/itwin/breakdown-trees/src/Views/ComponentIndex.tsx [hunk 19]
```diff
@@ -50,7 +50,7 @@ export const ComponentIndex: React.FC<ComponentIndexProps> = (props: ComponentIn
 
     populateContextMenuItems(treeName, functionIconMapperInner, dataProvider, props.eventHandlers);
 
-    const displayGuidHandlerInner = new GenericOptionItemHandler("Show Guids", BreakdownTrees.translate("contextMenu.showGuids"), "icon-label", () => { return props.displayGuids; }, props.setIsDisplayGuids);
+    const displayGuidHandlerInner = new GenericOptionItemHandler("Show Guids", BreakdownTrees.translate("contextMenu.showGuids"), "icon - label", () => { return props.displayGuids; }, props.setIsDisplayGuids);
```
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/Views/ComponentIndex.tsx:53]: why?

File: packages/itwin/breakdown-trees/src/Views/ClassificationsTree.tsx (hunk 18 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/Views/ClassificationsTree.tsx:49]: fixing that now, must have happened when i was doing a replace all by accident

File: packages/itwin/breakdown-trees/src/Views/SpatialContainmentTree.tsx [hunk 20]
```diff
@@ -101,11 +101,11 @@ export const SpatialContainmentTree: React.FC<SpatialContainmentTreeProps> = (pr
 
     populateContextAndOptionMenuItems(treeName, functionIconMapperInner, optionItemsInner, dataProvider, spatialRules, props.eventHandlers, props.clipHeight, props.clipAtSpaces);
 
-    const displayGuidHandlerInner = new GenericOptionItemHandler("Show Guids", BreakdownTrees.translate("contextMenu.showGuids"), "icon-label", () => props.displayGuids, props.setIsDisplayGuids);
+    const displayGuidHandlerInner = new GenericOptionItemHandler("Show Guids", BreakdownTrees.translate("contextMenu.showGuids"), "icon - label", () => props.displayGuids, props.setIsDisplayGuids);
```
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/Views/SpatialContainmentTree.tsx:104]: why are you adding spaces to every icon string?

File: apps/test-viewer/src/UiProvidersConfig.tsx (hunk 17 above)
Comment (Jake-Screen) [reply apps/test-viewer/src/UiProvidersConfig.tsx:79]: this is required for the MeasurementActionToolbar to be used in the viewer.

File: packages/itwin/breakdown-trees/src/BreakdownTrees.ts [hunk 21]
```diff
@@ -57,7 +62,12 @@ export class BreakdownTrees {
    * @internal
    */
   public static translate(key: string | string[], options?: LocalizationOptions): string {
-    return BreakdownTrees.i18n.getLocalizedStringWithNamespace(BreakdownTrees.i18nNamespace, key, options);
+    if (Array.isArray(key)) {
```
Comment (aruniverse) [inline packages/itwin/breakdown-trees/src/BreakdownTrees.ts:65]: is this right? i dont think your string concat does what you want it to. re-read what you wrote

File: packages/itwin/breakdown-trees/src/Views/SpatialContainmentTree.tsx (hunk 20 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/Views/SpatialContainmentTree.tsx:104]: i dont know why that happened, it maybe was a linter thing? i didnt do that myself

**Review by aruniverse — COMMENTED** [review]
more changes

File: packages/itwin/breakdown-trees/src/breakdown-trees-react.ts [hunk 22]
```diff
@@ -14,6 +14,7 @@ export * from "./Views/FunctionalityProviders/TreeNodeFunctionIconMapper";
 export * from "./Views/TreeWithRuleset";
 export * from "./Views/LoadableRuleSetComponent";
 export * from "./Views/OptionItemHandlers";
+export * from "./Views/SampleSpatialTree"
```
[the remark this answers is not in the document]
Comment (aruniverse) [reply packages/itwin/breakdown-trees/src/breakdown-trees-react.ts:17]: yes

File: packages/itwin/breakdown-trees/src/BreakdownTrees.ts (hunk 4 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/BreakdownTrees.ts]: changed

File: packages/itwin/breakdown-trees/src/test/Utils.tsx (hunk 6 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/test/Utils.tsx]: cleaned up

File: packages/itwin/breakdown-trees/src/test/Utils.tsx (hunk 7 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/test/Utils.tsx:58]: cleaned up

File: packages/itwin/breakdown-trees/src/test/functionalityProviders/BuildingClipPlanesProvider.test.ts (hunk 8 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/test/functionalityProviders/BuildingClipPlanesProvider.test.ts:50]: added `initialize` to the before part of the test

File: packages/itwin/breakdown-trees/src/test/functionalityProviders/SpaceClipPlanesProvider.test.ts (hunk 9 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/test/functionalityProviders/SpaceClipPlanesProvider.test.ts]: removed, was leftover from debugging

File: apps/test-viewer/src/components/Viewer.tsx (hunk 10 above)
Comment (Jake-Screen) [reply apps/test-viewer/src/components/Viewer.tsx]: removed

File: apps/test-viewer/src/components/Viewer.tsx (hunk 11 above)
Comment (Jake-Screen) [reply apps/test-viewer/src/components/Viewer.tsx]: changed

File: apps/test-viewer/.env.template [hunk 23]
```diff
@@ -19,4 +19,4 @@ TRANSPILE_DEPS=false
 
 # Space separated ids of widgets that should be added to viewer UI
 # Supported widget: 'tree-widget', 'property-grid'
-IMJS_ENABLED_WIDGETS = "tree-widget property-grid measure-tools"
\ No newline at end of file
+IMJS_ENABLED_WIDGETS = "tree-widget property-grid measure-tools breakdown-trees"
```
[the remark this answers is not in the document]
Comment (Jake-Screen) [reply apps/test-viewer/.env.template]: removed

File: apps/test-viewer/src/UiProvidersConfig.tsx [hunk 24]
```diff
@@ -44,12 +49,42 @@ interface UiItem {
   createUiItemsProvider: () => UiItemsProvider;
 }
 
+
+const SampleSpatialTree: React.FC = () => {
```
[the remark this answers is not in the document]
Comment (Jake-Screen) [reply apps/test-viewer/src/UiProvidersConfig.tsx]: removed to its own file in the components folder

File: apps/test-viewer/src/UiProvidersConfig.tsx (hunk 12 above)
Comment (Jake-Screen) [reply apps/test-viewer/src/UiProvidersConfig.tsx]: changed to inline

File: packages/itwin/breakdown-trees/src/breakdown-trees-react.ts (hunk 22 above)
[the remark this answers is not in the document]
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/breakdown-trees-react.ts]: moved to its own file in the test app

File: packages/itwin/breakdown-trees/src/Views/ClassificationsTree.tsx (hunk 18 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/Views/ClassificationsTree.tsx]: changed

File: packages/itwin/breakdown-trees/src/Views/ComponentIndex.tsx (hunk 19 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/Views/ComponentIndex.tsx]: changed

File: packages/itwin/breakdown-trees/src/Views/SpatialContainmentTree.tsx (hunk 20 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/Views/SpatialContainmentTree.tsx]: changed

File: packages/itwin/breakdown-trees/src/BreakdownTrees.ts (hunk 21 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/src/BreakdownTrees.ts:65]: fixed

File: packages/itwin/breakdown-trees/package.json [hunk 25]
```diff
@@ -99,22 +104,24 @@
...
-    "@itwin/presentation-frontend": "^3.0.0",
+    "@itwin/presentation-common": "^4.0.0-dev.66",
+    "@itwin/presentation-components": "^4.0.0-dev.30",
+    "@itwin/presentation-frontend": "^4.0.0-dev.66",
+    "@itwin/tree-widget-react": "workspace:*",
     "react": "^17.0.0",
     "react-dom": "^17.0.0",
-    "react-redux": "7.2.6"
+    "react-redux": "7.2.6",
+    "rxjs": "^6.6.2"
```
Comment (saskliutas) [inline packages/itwin/breakdown-trees/package.json:124]: Why is `rxjs` a peerDep of this package?

File: apps/test-viewer/src/UiProvidersConfig.tsx [hunk 26]
```diff
@@ -49,55 +46,39 @@ interface UiItem {
...
+          ),
+        }]
       }),
     }
   ],
   [
     "property-grid",
     {
-      initialize: async () => PropertyGridManager.initialize(),
+      initialize: async () => await PropertyGridManager.initialize(),
```
Comment (grigasp) [inline apps/test-viewer/src/UiProvidersConfig.tsx:71]: you don't need to await on promise if you're returning it

File: packages/itwin/breakdown-trees/src/Views/FunctionalityProviders/TreeNodeFunctionIconMapper.ts [hunk 27]
```diff
@@ -73,15 +74,20 @@ export class TreeNodeFunctionIconInfoMapper {
...
-      const elementKey = this._treeDataProvider.getNodeKey(node.item);
-      if (NodeKey.isGroupingNodeKey(elementKey)) {
-        returnedList.push(...this._groupNodeFunctionIconInfos);
-      } else if (NodeKey.isInstancesNodeKey(elementKey)) {
-        const classHierarchyArray = await IModelReadRpcInterface.getClient().getClassHierarchy(this._treeDataProvider.imodel.getRpcProps(), elementKey.instanceKeys[0].className);
-        for (const className of classHierarchyArray) {
-          const mappedFunctionalityProviders = this._TreeNodeClassFunctionIconInfoMap.get(className);
-          if (mappedFunctionalityProviders)
-            returnedList.push(...mappedFunctionalityProviders);
+      if (isPresentationTreeNodeItem(node.item)) {
```
Comment (grigasp) [inline packages/itwin/breakdown-trees/src/Views/FunctionalityProviders/TreeNodeFunctionIconMapper.ts:77]: nitpick - could instead use `node && isPresentationTreeNodeItem(node.item)` to reduce indentation

File: packages/itwin/breakdown-trees/src/Views/FunctionalityProviders/ZoomFunctionalityProvider.ts [hunk 28]
```diff
@@ -49,32 +50,38 @@ export class ZoomFunctionalityProvider extends TreeNodeFunctionalityProvider {
...
-      const data = await DataLink.querySpatialIndex(this._treeDataProvider.imodel, elementKey.instanceKeys[0].id);
-      if (data.length === 0) {
-        // check if element has any child with geometry
-        const child = await DataLink.queryChildWithGeometry(this._treeDataProvider.imodel, elementKey.instanceKeys[0].id);
-        if (child.length === 0) {
-          const message: NotifyMessageDetailsType = new NotifyMessageDetails(OutputMessagePriority.Info, BreakdownTrees.translate("BreakdownTrees:zoomToElement.briefTimeoutMessage"), BreakdownTrees.translate("BreakdownTrees:zoomToElement.detailedTimeoutMessage"), OutputMessageType.Toast);
-          MessageManager.addMessage(message);
-          return;
-        }
+    if (isPresentationTreeNodeItem(node.item)) {
```
Comment (grigasp) [inline packages/itwin/breakdown-trees/src/Views/FunctionalityProviders/ZoomFunctionalityProvider.ts:53]: nitpick - the condition could be inverted to reduce indentation:
```ts
if (!isPresentationTreeNodeItem(node.item)) {
  return;
}

// continue doing what you're doing
```

File: packages/itwin/breakdown-trees/package.json [hunk 29]
```diff
@@ -31,7 +31,8 @@
     "build:watch": "tsc -w",
     "clean": "rimraf lib",
     "//cover": "npm test",
-    "test": "mocha \"./lib/cjs/test/**/*.test.js\"",
+    "test": "",
+    "//test": "mocha \"./lib/cjs/test/**/*.test.js\"",
```
Comment (grigasp) [inline packages/itwin/breakdown-trees/package.json:35]: I suppose this will get reverted before completing the PR?

**Review by grigasp — DISMISSED** [review]

**Comment by saskliutas** [comment]
Opened PR updating tree-widget dependencies and fixing lint and test errors. https://github.com/iTwin/viewer-components-react/pull/442

File: apps/test-viewer/src/UiProvidersConfig.tsx (hunk 26 above)
Comment (grigasp) [reply apps/test-viewer/src/UiProvidersConfig.tsx]: @Jake-Screen, this particular line is now fine, but I see you also removed `await` where you shouldn't have.

You don't need `await` if you're returning a promise:
- either
  ```ts
  () => myPromiseFunction()
  ```
- or
  ```ts
  () => { return myPromiseFunction(); }
  ```

You do need an await if you're not returning, e.g.:
```ts
() => {
  await myPromiseFunction();
}
```

Comment (Jake-Screen) [reply apps/test-viewer/src/UiProvidersConfig.tsx]: Looking at that initialize function, i see that it does return a promise, so doesn't it not need a await in front of it right there?

Comment (Jake-Screen) [reply apps/test-viewer/src/UiProvidersConfig.tsx]: changed

**Review by grigasp — COMMENTED** [review]
LGTM, but leaving approval for @saskliutas and @aruniverse

**Review by saskliutas — APPROVED** [review]

File: rush.json [hunk 30]
```diff
@@ -18,6 +18,11 @@
     "defaultBranch": "master"
   },
   "projects": [
+    {
+      "packageName": "@itwin/breakdown-trees-react",
```
Comment (aruniverse) [inline rush.json:22]: change `version-policies.json` and bump the pkg's minor and set it to be a pre-release

File: packages/itwin/breakdown-trees/package.json [hunk 31]
```diff
@@ -31,37 +31,42 @@
...
+    "@itwin/appui-abstract": "^3.7.0",
+    "@itwin/appui-layout-react": "^4.0.0-dev.17",
+    "@itwin/appui-react": "^4.0.0-dev.17",
+    "@itwin/build-tools": "^3.7.0",
+    "@itwin/components-react": "^4.0.0-dev.17",
+    "@itwin/core-bentley": "^3.7.0",
+    "@itwin/core-common": "^3.7.0",
+    "@itwin/core-frontend": "^3.7.0",
+    "@itwin/core-geometry": "^3.7.0",
     "@itwin/core-i18n": "^3.0.0",
```
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:57]: core-i18n should also be ^3.7.0

File: packages/itwin/breakdown-trees/package.json [hunk 32]
```diff
@@ -73,6 +78,7 @@
     "@types/react-redux": "^7.1.9",
     "@types/sinon": "^7.5.1",
     "@wojtekmaj/enzyme-adapter-react-17": "^0.6.3",
+    "@reduxjs/toolkit": "1.9.3",
```
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:81]: alphabetize deps

File: packages/itwin/breakdown-trees/package.json [hunk 33]
```diff
@@ -99,19 +105,20 @@
...
+    "@itwin/appui-abstract": "^3.7.0",
+    "@itwin/appui-layout-react": "^4.0.0-dev.17",
+    "@itwin/appui-react": "^4.0.0-dev.17",
+    "@itwin/components-react": "^4.0.0-dev.17",
+    "@itwin/core-bentley": "^3.7.0",
+    "@itwin/core-common": "^3.7.0",
+    "@itwin/core-frontend": "^3.7.0",
+    "@itwin/core-geometry": "^3.7.0",
+    "@itwin/core-react": "^4.0.0-dev.17",
     "@itwin/measure-tools-react": "workspace:^0.10.6",
```
Comment (aruniverse) [inline packages/itwin/breakdown-trees/package.json:117]:
```suggestion
    "@itwin/measure-tools-react": "workspace:^0.11.0-dev.3",
```

File: rush.json (hunk 30 above)
Comment (Jake-Screen) [reply rush.json:22]: changed

File: packages/itwin/breakdown-trees/package.json (hunk 31 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/package.json]: changed

File: packages/itwin/breakdown-trees/package.json (hunk 32 above)
Comment (Jake-Screen) [reply packages/itwin/breakdown-trees/package.json]: reordered

**Review by aruniverse — APPROVED** [review]
